lib/common/memory.c

```diff
@@ -132,18 +133,18 @@ void h2o_mem_clear_pool(h2o_mem_pool_t *pool)
     }
     /* free chunks, and reset the first chunk */
     while (pool->chunks != NULL) {
-        struct st_h2o_mem_pool_chunk_t *next = pool->chunks->next;
+        union un_h2o_mem_pool_chunk_t *next = pool->chunks->next;
         h2o_mem_free_recycle(&mempool_allocator, pool->chunks);
         pool->chunks = next;
     }
-    pool->chunk_offset = sizeof(pool->chunks->bytes);
+    pool->chunk_offset = sizeof(*pool->chunks);
 }
 
-void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz)
+void *h2o_mem_alloc_pool_aligned(h2o_mem_pool_t *pool, size_t alignment, size_t sz)
 {
     void *ret;
 
-    if (sz >= sizeof(pool->chunks->bytes) / 4) {
+    if (sz >= sizeof(*pool->chunks) / 4) {
```

Comment:
Maybe this should be `sz >= (sizeof(pool->chunks->bytes) - sizeof(pool->chunks->next)) / 4`?

Reply:
yes.
we can change this if we want **exactly** the same as before.